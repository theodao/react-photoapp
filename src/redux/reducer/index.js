import { combineReducers } from 'redux';

export default combineReducers({
  auth: require('./authReducer').reducer,
});
