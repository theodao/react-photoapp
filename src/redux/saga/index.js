import { all } from 'redux-saga/effects';

/** All saga  */
import authSaga from './authSaga';
import categorySaga from './categorySaga';
import appSaga from './appSaga';

export default function* root() {
  yield all([...authSaga, ...categorySaga, ...appSaga]);
}
