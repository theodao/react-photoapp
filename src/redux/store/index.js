import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import logger from 'redux-logger';
import rootReducer from '../reducer';

export default () => {
  /** Redux configuration  */
  const middleware = [];
  const enhancers = [];

  /** Saga middleware  */
  const sagaMiddleware = createSagaMiddleware();
  middleware.push(sagaMiddleware);

  /** Logger middleware  */
  middleware.push(logger);

  enhancers.push(applyMiddleware(...middleware));

  /** Store configuration  */
  const store = createStore(rootReducer, compose(...enhancers));

  /** Run root saga  */
  // const sagaManager = sagaMiddleware.run();

  return {
    store,
    // sagaManager,
  };
};
