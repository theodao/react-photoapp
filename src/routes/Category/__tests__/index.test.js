import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { ItemList } from '../';

describe('Testing Item list component', () => {
  afterEach(cleanup);

  it('Should match snapshot', () => {
    const store = createStore(() => ({
      auth: {
        isLoggedIn: false,
      },
      app: {
        isLoading: false,
      },
    }));
    const { asFragment } = render(
      <Provider store={store}>
        <ItemList
          fetchItems={jest.fn()}
          auth={{
            userInformation: {},
          }}
          category={{
            isFetching: false,
            items: [],
          }}
        />
        , ,
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
