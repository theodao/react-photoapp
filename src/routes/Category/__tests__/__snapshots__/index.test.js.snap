// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Item list component Should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="display: flex; justify-content: space-between;"
    >
      <div
        style="margin-left: 10px;"
      >
        <div
          class="TopMenu u-backgroundWhite u-paddingVerticalExtraSmall"
        >
          <div
            class="TopMenu-list"
          >
            <div
              class="TopMenu-item u-positionRelative is-active u-flexInline u-alignItemsCenter u-cursorPointer"
            >
              <div
                class="TopMenu-itemBefore u-positionAbsolute u-positionLeft u-positionTop u-backgroundTransparent u-widthFull"
              />
              <div
                class="TopMenu-itemAfter u-positionAbsolute u-positionLeft u-positionBottom u-backgroundPrimary u-widthFull"
              />
              <span
                class="u-positionRelative u-flexInline u-flexGrow-1 u-paddingVerticalTiny hover:u-textDecorationNone u-textPrimary"
              >
                <div
                  class="u-flexGrow-1"
                >
                  Dashboard
                </div>
              </span>
            </div>
            <div
              class="TopMenu-item u-positionRelative u-marginLeftLarge u-flexInline u-alignItemsCenter u-cursorPointer"
            >
              <a
                class="u-positionRelative u-flexInline u-flexGrow-1 u-paddingVerticalTiny hover:u-textDecorationNone u-textDark hover:u-textPrimary"
                href="#"
                role="button"
              >
                <div
                  class="u-flexGrow-1"
                >
                  Add categories
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div>
        <button
          class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textWhite hover:u-textWhite u-backgroundPrimary hover:u-backgroundPrimaryDark u-border u-borderPrimary hover:u-textDecorationNone u-cursorPointer u-textUppercase"
          style="margin: 10px; font-size: 0.9rem;"
        >
          <span
            class="Button-label u-inlineBlock"
          >
            Login
          </span>
        </button>
        <button
          class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textGray hover:u-textGray u-backgroundLightest hover:u-backgroundUltraLight u-border hover:u-textDecorationNone u-cursorPointer u-textUppercase"
          style="margin: 10px; font-size: 0.9rem;"
        >
          <span
            class="Button-label u-inlineBlock"
          >
            Sign up
          </span>
        </button>
      </div>
    </div>
    <div>
      <div
        style="max-width: 1140px; display: flex; justify-content: center; margin: 0px auto;"
      >
        <div
          class="EmptyState u-inlineBlock u-textCenter"
        >
          <img
            alt="General"
            class="u-maxWidthFull"
            src="https://designsystem-cdn.got-it.ai/images/general.8da83c28.svg"
            width="240"
          />
          <div
            class="EmptyState-heading u-marginTopSmall u-text650 u-textLight"
          >
            Item Empty
          </div>
        </div>
      </div>
    </div>
  </div>
  , ,
</DocumentFragment>
`;
